### ISSUE TITLE
gubernator SSL client auth not working as expected

### ISSUE THREAD

**Issue by jjmanzer** [mailgun/gubernator#122]
Thank you for writing this really cool opensource tool.

We are attempting to use gubernator for our own rate limiting purposes and ran into an issue with client SSL authentication. This is on clustered 2 gubernator instances running behind a load balancer. The SSL cert has a name of the vip ($fqdn). When one tries to connect to the VIP with a POST against `https://$fqdn/v1/GetRateLimits` the following is returned...

```
{
    "responses": [
        {
            "status": "UNDER_LIMIT",
            "limit": "0",
            "remaining": "0",
            "reset_time": "0",
            "error": "while fetching rate limit 'requests_per_sec_abc' from peer - 'rpc error: code = Unavailable desc = connection error: desc = \"transport: authentication handshake failed: x509: cannot validate certificate for 10.26.42.0 because it doesn't contain any IP SANs\"'",
            "metadata": {}
        }
    ]
}
```

The environment is as follows
```
      - GUBER_HTTP_ADDRESS: 0.0.0.0:8080
      - GUBER_GRPC_ADDRESS: 0.0.0.0:8081
      - GUBER_ADVERTISE_ADDRESS: 10.26.42.0:8081
      - GUBER_MEMBERLIST_KNOWN_NODES: $vm1_name,$vm2_name2
      - GUBER_TLS_CA: path_to_intermediate_chain_crt
      - GUBER_TLS_KEY: path_to_key
      - GUBER_TLS_CERT: path_to_crt
      - GUBER_TLS_CLIENT_AUTH: require-any-cert
```

The 10.26.42.0 address mentioned in the error is the directly bound IP address for my VM that the service is running on but it should not be authenticating against that address, The app should be authenticating against the name (the $fqdn) provided in the HTTPS request I think. Is this a bug or am I doing something wrong?

**Comment by thrawn01** [comment]
Hi @jjmanzer, that error is coming from one of the other peers in the cluster. With your current config both peer to peer and client connections are using the same TLS certs. You might consider using separate TLS certs for client vs peer communication or only use client TLS.  Looking back now, I think including the `_AUTH` in `GUBER_TLS_CLIENT_AUTH_KEY` was probably confusing.

try this.
```

**Comment by jjmanzer** [comment]
When I did this I got:
```
 [DEBUG] memberlist: Initiating push/pull sync with:  x.x.x.x:7946" category=gubernator
Jan 24 14:57:46 a2tlnetlimit01 journal: time="2022-01-24T21:57:46Z" level=info msg="2022/01/24 21:57:46 [DEBUG] memberlist: Stream connection from=172.17.0.1:41918" category=gubernator
Jan 24 14:57:46 a2tlnetlimit01 journal: time="2022-01-24T21:57:46Z" level=info msg="2022/01/24 21:57:46 [DEBUG] memberlist: Initiating push/pull sync with:  y.y.y.y:7946" category=gubernator
Jan 24 14:57:46 a2tlnetlimit01 journal: time="2022-01-24T21:57:46Z" level=info msg="HTTPS Gateway Listening on 0.0.0.0:8080 ..." category=gubernator
Jan 24 14:57:46 a2tlnetlimit01 journal: time="2022-01-24T21:57:46Z" level=error msg="while starting TLS HTTP server" category=gubernator error="open : no such file or directory"
```
The cert files are owned by root and the docker container runs as root and the cert dir is bound to the container so the files should be reachable. Btw I am using salt's docker module to drive the docker container logic if that matters.
On the subject of using multiple certs, I would like to keep it to one wildcard cert if I can as that is easier to manage. I would like to encrypt both client and cluster sessions if possible. But even with the updated env I got the above errors in /var/log/messages.

**Comment by thrawn01** [comment]
@denkyl08 A PR for this would be VERY welcome, Thank you for your other PR's as well, they are much appreciated.

**Comment by thrawn01** [comment]
looks like might be resolved by #133. I'm closing this issue.
